python/3d/auto_generated/materials/qgsphongtexturedmaterialsettings.sip.in

```diff
@@ -65,11 +72,11 @@ Returns shininess of the surface
     virtual QMap<QString, QString> toExportParameters() const;
 
 
-    QString diffuseTexturePath() const;
+    QVariant diffuseTexture() const;
 %Docstring
-Returns the diffuse texture path.
+Returns the diffuse texture component of the material. This can be either a path as a QString or a texture.
```

Comment:
Can you make a note as to what value types are acceptable for a texture? Eg is a QImage acceptable?

Reply:
Done.